androidApp/src/main/java/com/mbta/tid/mbta_app/android/component/TransitHeader.kt

```diff
@@ -44,6 +46,7 @@ fun TransitHeader(
             Text(
                 text = name,
                 maxLines = 1,
+                modifier = Modifier.semantics { heading() },
```

Comment:
Is there an easy rule for when we'll need to add custom semantics?

Reply:
Not quite an easy rule, but when the default behavior doesn't match the designs or impedes the user's ability to find and understand the information. Any visual cues that a sighted user gets from using the app should be represented via semantics (such as heading level here). When we are using built-in material components, we shouldn't have to worry about custom semantics unless we want to override the defaults to be more descriptive for that particular element in its context.

For the most part this PR is directly duplicating the customizations done on iOS, though there are some differences (iOS supports heading levels, whereas heading is binary on android).